src/pal/tools/gen-buildsys-win.bat

```diff
@@ -18,18 +18,28 @@ if %basePath:~-1%==\ set "basePath=%basePath:~0,-1%"
 set __SourceDir=%1
 set __VSVersion=%2
 set __Arch=%3
+set __HostToolset=%4
 set __CmakeGenerator=Visual Studio
 if /i "%__VSVersion%" == "vs2017" (set __CmakeGenerator=%__CmakeGenerator% 15 2017)
 if /i "%__VSVersion%" == "vs2015" (set __CmakeGenerator=%__CmakeGenerator% 14 2015)
 if /i "%__Arch%" == "x64" (set __CmakeGenerator=%__CmakeGenerator% Win64)
 if /i "%__Arch%" == "arm64" (set __CmakeGenerator=%__CmakeGenerator% Win64)
 if /i "%__Arch%" == "arm" (set __CmakeGenerator=%__CmakeGenerator% ARM)
 
+set __HOST=""
+
+echo %__HostToolset%
+
+if /i %__HostToolset% == "64" (
```

Comment:
You're expecting the argument to come in surrounded by quotes. That is very weird. I wouldn't do that.

Reply:
Since you're specifically checking the argument, why not make it more explicit, like "UseX64HostedTools" or "UseX86HostedTools"?